pallets/anchor/src/lib.rs

```diff
@@ -352,16 +352,17 @@ impl<T: Config<I>, I: 'static> AnchorInterface<AnchorConfigration<T, I>> for Pal
 		let refund_bytes = refund.using_encoded(element_encoder);
 		let chain_id_bytes = chain_id.using_encoded(element_encoder);
 
+		bytes.extend_from_slice(&chain_id_bytes);
 		bytes.extend_from_slice(&nullifier_hash.encode());
+		for root in &roots {
+			bytes.extend_from_slice(&root.encode());
+		}
+		bytes.extend_from_slice(&roots[0].encode());
```

Comment:
Can you explain this and document? Why are we adding the first root again at the end?